openmc/cppapi/cppapi.cpp

```diff
@@ -14,6 +14,15 @@ namespace openmc {
 using std::string;
 using std::vector;
 
+// General exception handler 
+// This function is called by openmc_cppapi functions
+// that wrap error codes, and throw exceptions
+void exception_handler(string func, int err) {
+  string msg = "function " + func + " exited with error code: "
```

Comment:
When our C API functions return an error code, there's actually a global error string `openmc_err_msg` that stores the message. If you look at our current error handler in Python for the C API, you'll see we use `openmc_err_msg` to see the message of the exception. The same thing should be done here.